05-rx-js/src/services/utils.js

```diff
@@ -0,0 +1,31 @@
+import { HttpService as httpService } from "./http.service";
+import { map } from "rxjs/operators";
+
+export function loadPosts() {
+    return httpService.getData('https://jsonplaceholder.typicode.com/posts').pipe(
+        map((data) => convertPostsToMarkup(data))
+    );
+}
+
+export function filterBySelector(query, parentElem) {
+    parentElem.innerHTML = '';
+    httpService.getData('https://jsonplaceholder.typicode.com/posts').pipe(
+        map((data) => data.filter(
+            post => post.title.includes(query)
+        )),
+        map((data) => convertPostsToMarkup(data))
+    ).subscribe((markUp) => parentElem.innerHTML = markUp);
```

Comment:
I don't think you were supposed to re-load the data on filtering:

```
5p. Create input. Wrap input changes in subject. On every input change wait until no new data
comes in 1 sec then search in posts (from previous 2 steps) body and filter UI.
```
It ways nowhere you should re-load the data. It says "search in posts (from previous 2 steps)" meaning you should search in what was already loaded.

Reply:
Gerenally this function does too much of different things and thus it's name is confusing. It is called `filterBySelector`, so one would expect it to filter by selector. But what it does instead is:
 - cleans the parent element innerHTML
 - loads the data again with HTTP service, ignoring what was already loaded
- calls the convertion to markup
- inserts the markup

Not exactly the implementation of the Separation of concerns principle.

Also having the URL hardcoded in two different functions should've already be a red flag. This duplication is a code smell indicating something is being done wrong (even if it seems to work).